first_hour_on_terra/workspace_resource_examples.ipynb

```diff
@@ -0,0 +1,1552 @@
+    "            button_style='',\n",
+    "            tooltip='Click to add a GCS bucket as a referenced resource.',\n",
+    "            icon='plus'\n",
+    "        )\n",
+    "        self.vb = widgets.VBox(\n",
+    "            [\n",
+    "                self.label,\n",
+    "                self.cloning_drop_down,\n",
+    "                self.input_bucket_name,\n",
+    "                self.input_description,\n",
+    "                self.input_name,\n",
+    "                self.button,\n",
+    "                self.output\n",
+    "            ]\n",
+    "        )\n",
+    "        self.button.on_click(self.add_gcs_bucket)\n",
+    "\n",
+    "    def add_gcs_bucket(self, b):\n",
+    "        with self.output:\n",
+    "            description_content = f\"\\\"{self.input_description.value}\\\"\"\n",
+    "\n",
+    "            terraCommand = f\"\"\"terra resource add-ref gcs-bucket \\\\\n",
+    "            --cloning={self.cloning_drop_down.value} \\\\\n",
+    "            --bucket-name={self.input_bucket_name.value} \\\\\n",
+    "            --description={description_content} \\\\\n",
+    "            --name={self.input_name.value} \\\\\n",
+    "            --workspace={CURRENT_WORKSPACE_ID}\n",
+    "            \"\"\"\n",
+    "            print(terraCommand)\n",
+    "            result = subprocess.run([\"terra\", \"resource\", \"add-ref\", \"gcs-bucket\",\n",
+    "                                     f\"--cloning={self.cloning_drop_down.value}\",\n",
+    "                                     f\"--bucket-name={self.input_bucket_name.value}\",\n",
+    "                                     f\"--name={self.input_name.value}\",\n",
+    "                                     f\"--description={description_content}\",\n",
+    "                                     f\"--workspace={CURRENT_WORKSPACE_ID}\"], capture_output=True, text=True)\n",
+    "            print(result.stderr) if not result.stdout else print(result.stdout)\n",
+    "\n",
+    "\n",
+    "# Instantiate widget\n",
+    "add_gcs_bucket_widget = AddGcsBucketWidget()\n",
+    "display(add_gcs_bucket_widget.vb)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 5. Add Google Cloud Storage object\n",
+    "\n",
+    "Run the cell below to create a widget, then populate the widget's input fields and click the button to add a Google Cloud Storage bucket object as a referenced resource in your workspace.\n",
+    "\n",
+    "Widget input parameters include:\n",
+    "\n",
+    "- `Cloning`: _You cannot change the workspace ID after workspace creation._\n",
+    "- `Bucket Name`: Must be a string that [meets requirements](https://cloud.google.com/storage/docs/buckets#naming) and is globally unique. We suggest prepending the `<PROJECT_ID>` of the GCP project associated with your workspace, which can be found on the to a descriptive string (e.g., `<PROJECT_ID>_output_data`).\n",
```

Comment:
Grammar at end of this sentence, same issue as above

Reply:
fixed